### PR TITLE
ZP - Driver Availability Table and Tests

### LINKED ISSUE #13: Driver Availability #4: Add DriverAvailabilityTable
Prerequisites #10 #11, part of #4 

### 4th issue: DriverAvailabilityTable

- [x] There is a react component `DriverAvailabilityTable` that is suitable for use on the pages described in the acceptance criteria
- [x] There are storybook entries

### PR DESCRIPTION
**Pull request by Ze-el**
In this PR, I added the Driver Availability Table, Stories and Utils. It also covers all tests that have 100%  line and mutation coverage. The driver table has edit and delete buttons but you need the role of driver to check. Similarly with admin table there is a review button but you need the role of a admin to check.

Storybook Link: https://ucsb-cs156-w24.github.io/proj-gauchoride-w24-7pm-1/prs/45/storybook/?path=/story/components-chatdisplay--default

Screenshot 2024-03-06 at 10 14 07 PM
Screenshot 2024-03-06 at 10 14 12 PM
Screenshot 2024-03-06 at 10 14 17 PM
Screenshot 2024-03-06 at 10 14 22 PM

Closes #13

### CODE REVIEW

**Review by gretchenlam — DISMISSED** [review]
LGTM!

**Review by liamlizard — DISMISSED** [review]
LGTM! Works well on storybook

**Review by jamespflaging — DISMISSED** [review]
LGTM

File: frontend/src/main/components/Driver/DriverAvailabilityTable.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            Header: 'startTime',
+            accessor: 'startTime',
+        },
+        {
+            Header: 'endTime',
+            accessor: 'endTime',
+        },
+        {
+            Header: 'notes',
+            accessor: 'notes',
```
Comment (andrewpengucsb) [inline frontend/src/main/components/Driver/DriverAvailabilityTable.js:58]: Would like to see some nicer headers for the table aka 

Driver ID
Start Time
etc

**Review by andrewpengucsb — CHANGES_REQUESTED** [review]

File: frontend/src/main/components/Driver/DriverAvailabilityTable.js (hunk 1 above)
Comment (Ze-el) [reply frontend/src/main/components/Driver/DriverAvailabilityTable.js]: Got it!

**Comment by Ze-el** [comment]
Hey Andrew, I have changed the headers to make them much more understanding and nicer like you requested.

**Review by jamespflaging — APPROVED** [review]
new headers look good!

**Review by andrewpengucsb — APPROVED** [review]
LGTM

**Pull request merged**